### ISSUE TITLE
Feature request: Array property extending doesn't work when property has default value

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#2228]
PR #2154 added support for array properties being extended, however it appears that this doesn't work when an array property already has a default value set.

If we take the original example of the `Generic.PHP.ForbiddenFunctions` sniff, it all works correctly if you overrule the `forbiddenFunctions` property in a standard and then `extend` it from a project-based ruleset.

However, if you want to extend the default values of the `forbiddenFunctions` property:
```php
public $forbiddenFunctions = [
     'sizeof' => 'count',
     'delete' => 'unset',
];
```

either via a standard or a project ruleset:
```xml
<rule ref="Generic.PHP.ForbiddenFunctions">
	<properties>
		<property name="forbiddenFunctions" type="array" extend="true">
			<element key="foo" value="fixedFoo"/>
			<element key="bar" value="fixedBar"/>
		</property>
	</properties>
</rule>
```

You end up with:
```php
public $forbiddenFunctions = [
    'foo' => 'fixedFoo',
    'bar' => 'fixedBar',
];
```

instead of the expected:
```php
public $forbiddenFunctions = [
    'sizeof' => 'count',
    'delete' => 'unset',
    'foo' => 'fixedFoo',
    'bar' => 'fixedBar',
];
```

@Majkl578 Any thoughts on this ?

Related #2153

**Comment by gsherwood** [comment]
This is an interesting use case, but not what the feature was designed to do, or how I documented it. I don't think this should be considered a bug - it just a feature that *still* doesn't exist despite the `extends` feature being added.
